src/common/io/socket/common.c

```diff
@@ -126,3 +128,86 @@ sckOptionSet(int fd)
 
     FUNCTION_TEST_RETURN_VOID();
 }
+
+/***********************************************************************************************************************************
+Use poll() to determine when data is ready to read/write on a socket. Retry after EINTR with whatever time is left on the timer.
+***********************************************************************************************************************************/
+// Helper to determine when poll() should be retried
+static bool
+sckReadyRetry(int pollResult, int errNo, bool first, Wait *wait)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM(INT, pollResult);
+        FUNCTION_TEST_PARAM(INT, errNo);
+        FUNCTION_TEST_PARAM(BOOL, first);
+        FUNCTION_TEST_PARAM(WAIT, wait);
+    FUNCTION_TEST_END();
```

Comment:
Can "Wait *wait" be NULL? If not, should we have an ASSERT?

Reply:
This parameter has been removed, but I did add asserts for the new params.